cmd/restic/cmd_check.go

```diff
@@ -287,30 +316,48 @@ func runCheck(opts CheckOptions, gopts GlobalOptions, args []string) error {
 		doReadData(selectPacksByBucket(chkr.GetPacks(), 1, 1))
 	case opts.ReadDataSubset != "":
 		var packs map[restic.ID]int64
-		dataSubset, err := stringToIntSlice(opts.ReadDataSubset)
+		readDataSubset, err := determineReadDataSubsetType(opts.ReadDataSubset)
 		if err == nil {
-			bucket := dataSubset[0]
-			totalBuckets := dataSubset[1]
-			packs = selectPacksByBucket(chkr.GetPacks(), bucket, totalBuckets)
-			packCount := uint64(len(packs))
-			Verbosef("read group #%d of %d data packs (out of total %d packs in %d groups)\n", bucket, packCount, chkr.CountPacks(), totalBuckets)
-		} else {
-			percentage, _ := parsePercentage(opts.ReadDataSubset)
-			packs = selectRandomPacksByPercentage(chkr.GetPacks(), percentage)
-			Verbosef("read %.1f%% of data packs\n", percentage)
+			switch readDataSubset {
+			case Percentage:
+				percentage, err := parsePercentage(opts.ReadDataSubset)
+				if err == nil {
+					packs = selectRandomPacksByPercentage(chkr.GetPacks(), percentage)
+					Verbosef("read %.1f%% of data packs\n", percentage)
+				}
+			case FileSize:
+				repoSize, err := dirSize(gopts.Repo)
+				if err != nil {
+					errors.Fatal("Could not get the file size of repo")
+				}
+				subsetSize, _ := parseSizeStr(opts.ReadDataSubset)
+				if subsetSize > repoSize {
+					subsetSize = repoSize
+				}
+				packs = selectRandomPacksByFileSize(chkr.GetPacks(), subsetSize, repoSize)
+				Verbosef("read %d of data packs\n", subsetSize)
+			case Fraction:
+				dataSubset, err := stringToIntSlice(opts.ReadDataSubset)
+				if err == nil {
+					bucket := dataSubset[0]
+					totalBuckets := dataSubset[1]
+					packs = selectPacksByBucket(chkr.GetPacks(), bucket, totalBuckets)
+					packCount := uint64(len(packs))
+					Verbosef("read group #%d of %d data packs (out of total %d packs in %d groups)\n", bucket, packCount, chkr.CountPacks(), totalBuckets)
+				}
+			default:
+				errors.Fatal("Invalid value for read-data-subset")
+			}
 		}
 		if packs == nil {
 			return errors.Fatal("internal error: failed to select packs to check")
 		}
 		doReadData(packs)
 	}
-
```

Comment:
Why remove these lines? I think it's better to only touch what's necessary for your change.